dask/array/lib/stride_tricks.py

```diff
@@ -0,0 +1,71 @@
+from ..overlap import rechunk_for_overlap, map_overlap, trim_chunks
+
+
+def _window_view(arr, window_shape, axis, fix_first_block, block_info):
+    """ core sliding window function"""
+    from ..numpy_compat import sliding_window_view
+
+    block_info = block_info[0]
+    slicer = [slice(None, None)] * arr.ndim
+    for ax, window, fix_first in zip(axis, window_shape, fix_first_block):
+        start, stop = None, None
+        is_last_chunk = (
+            block_info["num-chunks"][ax] - 1 - block_info["chunk-location"][ax]
+        ) == 0
+
+        locs = block_info["array-location"][ax]
+        size = locs[1] - locs[0]
+        if not is_last_chunk:
+            stop = -1
+
+        # when depth == chunk size of first chunk along some axis
+        # we need to make some confusing empirical adjustments.
+        # could be avoided if we rechunked so that min
+        # chunk size is depth+1 (which would also be more efficient)
+        # 1) don't trim the first block along that axis
+        if size == window and block_info["chunk-location"][ax] == 0:
+            stop = None
+        # 2) we need to skip the first "row" of the second block
+        if fix_first and block_info["chunk-location"][ax] == 1:
+            start = 1
+        slicer[ax] = slice(start, stop)
+
+    view = sliding_window_view(arr[tuple(slicer)], window_shape, axis)
+    return view
+
+
+def sliding_window_view(arr, window_shape, axis):
+    """ Dask wrapper for numpy.lib.stride_tricks.sliding_window_view."""
+
+    ax_wind = {ax: (window, 0) for ax, window in zip(axis, window_shape)}
```

Comment:
Is this possibly supposed to be `(0, window)` rather than `(window, 0)`? I haven't used `map_overlap` myself so I might have the convention wrong, but it seems that by doing it this way around, it's the first chunk that will shrink (because there is no overlap from the left), but you're trimming the last chunk size.